docs/configuration.md

```diff
@@ -31,6 +53,24 @@ extensions:
       namespace: default
 ```
 
+### Configuration Validation
+
+k0s command-line interface has the ability to validate config syntax:
+
+```
+$ k0s validate config --config path/to/config/file
+```
+
+`validate config` sub-command can validate the following:
+
+1. YAML formatting
+2. [SANs addresses](#specapi-1)
+3. [Network providers](#specnetwork-1)
+4. [Worker profiles](#specworkerprofiles) 
+
+
+## Configuration Keys
+
 ### `spec.api`
 
```

Comment:
It looks like a we have two sections describing spec.api, spec.network and helm extensions. one under the "configuration keys" and also under "full configuration". 
It's not clear why we have both, so I would merge the configuration keys and the full configuration sections (and rename "full configuration" to "configuration keys")

Reply:
we also have two "example" outputs, which I believe is also redundant.

Reply:
The latter one if for HA setup. Maybe it should go into a separate file?

Reply:
We can keep the one for HA setup, since it's the more "production-ready" setup.